File: Problems/No.66_Plus_One/plus_one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        
        # convert digits into integer
        digits_int = int("".join(str(i) for i in digits))
        
        # convert the integer result after plus one into list
        PO_ans = list(map(int, str(digits_int+1)))
        
        return PO_ans

File: Problems/No.66_Plus_One/plus_one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        
        // check if the carry occurs
        bool carry = 0;
        
        // start compute
        for (int i=digits.size()-1; i>=0; i--) {
            
            digits[i] += 1;
            if (digits[i] >= 10) {
                
                carry = 1;
                digits[i] -= 10;
                
            }
            else {
                
                carry = 0;
                break;
                
            }
            
        }
        
        if (carry) digits.insert(digits.begin(), 1);
        return digits;
        
    }
};
